Add self-hosting recommendations

nydr opened the pull request:
- Adds table for self-hosting recommendations/consideration
- New shortcodes `icon/yes`, `icon/no`, `format-number`

nydr commented on content/en/apps/guides/hosting/4.x/self-hosting/_index.md:
```diff
@@ -0,0 +1,39 @@
+| ------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------- |
+| Less than 2000 users                                                            | {{< icon/yes >}}                                                                         | {{< icon/yes >}}                                                                                     |
+| More than 2000 users                                                            | {{< icon/no >}}                                                                          | {{< icon/yes >}}                                                                                     |
+| Less than 1 million documents                                                   | {{< icon/yes >}}                                                                         | {{< icon/yes >}}                                                                                     |
```
@garethbowen do you know if these numbers are accurate?

nydr commented on content/en/apps/guides/hosting/4.x/self-hosting/_index.md:
```diff
@@ -0,0 +1,39 @@
+| Less than 2000 users                                                            | {{< icon/yes >}}                    | {{< icon/yes >}}                            |
+| More than 2000 users                                                            | {{< icon/no >}}                     | {{< icon/yes >}}                            |
+| Less than 1 million documents                                                   | {{< icon/yes >}}                    | {{< icon/yes >}}                            |
+| More than 1 million documents                                                   | {{< icon/no >}}                     | {{< icon/yes >}}                            |
```
@garethbowen do you know if these numbers are accurate?

garethbowen replied:
I think it would be more important to think about docs per day not just outright doc count. If a db has a lot of docs but it's not growing much then single node is probably fine. I think the tipping point would be somewhere around 10k docs in a day. This accounts for the burst traffic point below...

The user count is probably a little low - we have single nodes working with ~4k users right now. Let's set it to 4k.

garethbowen commented on content/en/apps/guides/hosting/4.x/self-hosting/_index.md:
```diff
@@ -0,0 +1,39 @@
+| More than 2000 users                                                            | {{< icon/no >}}                     | {{< icon/yes >}}                            |
+| Less than 1 million documents                                                   | {{< icon/yes >}}                    | {{< icon/yes >}}                            |
+| More than 1 million documents                                                   | {{< icon/no >}}                     | {{< icon/yes >}}                            |
+| Significant bursts of traffic (i.e. "Sync day" for a lot of users once a month) | {{< icon/no >}}                     | {{< icon/yes >}}                            |
```
"sync day" applies to most instances so this would rule out single node for just about everyone. I think we should remove this line and focus on the docs/day measure instead.

garethbowen commented on content/en/apps/guides/hosting/4.x/self-hosting/_index.md:
```diff
@@ -0,0 +1,39 @@
+| --------------------------- | --------------- | ---------------- |
+| Data needs to be in-country | {{< icon/no >}} | {{< icon/yes >}} |
+
+## Recommended orchestration provider
+
+| Consideration                        | Docker Compose   | K3s              |
+| ------------------------------------ | ---------------- | ---------------- |
+| Small deployment (less than 3 nodes) | {{< icon/yes >}} | {{< icon/yes >}} |
+| Large deployment                     | {{< icon/no >}}  | {{< icon/yes >}} |
```
I think we should move away from recommending docker-compose in production even for small nodes. It's fine for demos, testing, and dev but not prod. I know the documentation isn't quite there yet but let's not suggest that compose is ever ok.

garethbowen reviewed (changes requested):
Amazing work - thanks!

I've put suggestions inline.

If it's not too much work could this be split into two PRs? One for build updates, and one for the content?

nydr commented:
> could this be split into two PRs?

PR updated and build updates moved to https://github.com/medic/cht-docs/pull/1198

garethbowen commented on content/en/apps/guides/hosting/4.x/self-hosting/_index.md:
```diff
@@ -0,0 +1,31 @@
+
+| Consideration                                   | [Single node CouchDB][single-couch] | [Multi-node clustered CouchDB][multi-couch] |
+| ----------------------------------------------- | ----------------------------------- | ------------------------------------------- |
+| Less than 4000 users                            | {{< icon/yes >}}                    | {{< icon/yes >}}                            |
+| More than 4000 users                            | {{< icon/no >}}                     | {{< icon/yes >}}                            |
+| Less than 10 000 documents per day              | {{< icon/yes >}}                    | {{< icon/yes >}}                            |
+| More than 10 000 documents per day              | {{< icon/no >}}                     | {{< icon/yes >}}                            |
```
Nitpick: for consistency the "4000" number should have a thousand separator too. I also prefer commas to spaces because they don't line wrap, but it's also different in different locales so I'm not too worried about it.

```suggestion
| Less than 4,000 users                            | {{< icon/yes >}}                    | {{< icon/yes >}}                            |
| More than 4,000 users                            | {{< icon/no >}}                     | {{< icon/yes >}}                            |
| Less than 10,000 documents per day              | {{< icon/yes >}}                    | {{< icon/yes >}}                            |
| More than 10,000 documents per day              | {{< icon/no >}}                     | {{< icon/yes >}}                            |
```

nydr replied:
While I agree that comma is generally easier to read, especially for english speaking countries, I want to insist on avoiding them in international documentation as some locales use comma as decimal separator. I went with spaces seem to be the most accepted international standard [ref](https://en.wikipedia.org/wiki/Decimal_separator#Digit_grouping) and there's no risk of confusing it with a decimal separator.

I updated the PR to use the unicode character `Thin Space` which is thinner than a regular space in most fonts and should not word wrap.

Other options would be formatting numbers based on the browser locale using client side javascript, and/or using css for spacing to assist in functional copy paste, but I think that's overkill for this.

garethbowen replied:
> While I agree that comma is generally easier to read, especially for english speaking countries, I want to insist on avoiding them in international documentation as some locales use comma as decimal separator. I went with spaces as they seem to be the most accepted international standard [[ref](https://en.wikipedia.org/wiki/Decimal_separator#Digit_grouping)] and there's no risk of confusing it with a decimal separator.

Good point, well made! Would you mind updating the [style guide](https://docs.communityhealthtoolkit.org/contribute/docs/style-guide/) with this so we can stay consistent?

> Other options would be formatting numbers based on the browser locale using client side javascript, and/or using css for spacing to assist in functional copy paste, but I think that's overkill for this.

It seems crazy to me that there isn't a standard browser feature for this. Dates are even worse.

nydr replied:
> Would you mind updating the [style guide](https://docs.communityhealthtoolkit.org/contribute/docs/style-guide/) with this so we can stay consistent?

Sure, I'll open a new PR for this

@mrjones-plip this PR is blocked due to "changes requested", please give it a look when you have a moment

garethbowen reviewed (approved):
Great work, thanks!

One optional nitpick inline.

garethbowen reviewed (approved):
Good to go!

mrjones-plip reviewed (approved):
approving to unblock the PR, but noting the title and body are still empty/need updating per my initial request below.  You can fix this on the next PRs!

> but can you update the title of this PR and put a body in it listing out what the changes/improvements are? For example, it currently mentions self hosting, but then has extensive changes to [GH CI files](https://github.com/medic/cht-docs/pull/1191/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR25) and [readme](https://github.com/medic/cht-docs/pull/1191/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R13) etc.

nydr commented:
> approving to unblock the PR, but noting the title and body are still empty/need updating per my initial request below. You can fix this on the next PRs!

Thanks @mrjones-plip ! I figured it wasn't needed since I split up the PR into multiple parts but I added a short description just now, is this what you were after?
